packages/jest-editor-support/src/SnapshotSupport.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+const babylon = require('babylon');
+const fs = require('fs');
+const path = require('path');
+const walker = require('./walker');
+
+type Node = any;
+
+type SnapshotMetadata = {
+  exists: true | false,
+  exportName: string,
+  node: Node,
+  content?: string,
+};
+
+type SnapshotSupportType = {
+  getMetadataFromTestFile: (filePath: string) => Array<SnapshotMetadata>,
+  parser?: (content: string, options: any) => Node,
+  plugins?: Array<string>,
+};
+
+const buildExportName: (
+  toMatchSnapshot: Node,
+  parents: Array<Node>
+) => string = (toMatchSnapshot, parents) => {
+  const state = {
+    found: [],
+  };
+
+  walker(
+    parents[parents.length - 1],
+    {
+      Identifier(node, state, parents) {
+        if (node.name === 'toMatchSnapshot') {
+          state.found.push({node, parents});
+        }
+      },
+    },
+    state
+  );
```

Comment:
The walker already does this, parents is an array of the parents, and I use that to select the nearest `it` or `test` parent.
Here though we would need to check for all `toMatchSnapshots` if they have a common ancestor before starting this, and group by it. 
Since I already filter the parents, it should be just a matter of moving the `parents.filter` up inside the map and then comparing just the last parent (and I can even optimize this last bit by keeping a reference to the last parent and reset the index when it change or increment it)